### PR TITLE
Refactor workarounds in application tests

### PR DESCRIPTION
**Pull request by taehoonlee**
This PR follows up #9022. The multiprocessing calls have been factorized with a common function `clear_run`.

### CODE REVIEW

File: tests/keras/applications/applications_test.py [hunk 1]
```diff
@@ -8,6 +8,27 @@
...
 
+def clean_run(model_fn):
+    if K.backend() == 'cntk':
+        # Create model in a subprocess so that the memory consumed by InceptionResNetV2 will be
+        # released back to the system after this test (to deal with OOM error on CNTK backend)
+        # TODO: remove the use of multiprocessing from these tests once a memory clearing mechanism
+        # is implemented in the CNTK backend
+        def target(queue):
+            model = model_fn()
+            queue.put(model)
```
Comment (myutwo150) [inline tests/keras/applications/applications_test.py]: I don't think we can pickle Keras models back to the main process. That's why only the output shape is put into the queue in the original lines.

**Comment by fchollet** [comment]
@myutwo150 thanks for reviewing. Will wait for a concerted LGTM between you two before merging.

File: tests/keras/applications/applications_test.py (hunk 1 above)
Comment (taehoonlee) [reply tests/keras/applications/applications_test.py]: Updated. Thank you for the comments @myutwo150.

File: tests/keras/applications/applications_test.py [hunk 2]
```diff
@@ -175,87 +196,45 @@ def test_inceptionv3_variable_input_channels():
...
+        return applications.InceptionResNetV2(weights=None, include_top=False)
     global_image_data_format = K.image_data_format()
-    queue = Queue()
-
     K.set_image_data_format('channels_first')
-    p = Process(target=target, args=(queue,))
-    p.start()
-    p.join()
+    model = clean_run(model_fn)
     K.set_image_data_format(global_image_data_format)
```
[the remark this answers is not in the document]
Comment (taehoonlee) [reply tests/keras/applications/applications_test.py]: I agree. How about removing `set_image_data_format`? Other application tests except `test_inceptionresnetv2_notop` do not include `set_image_data_format`.

[the remark this answers is not in the document]
Comment (taehoonlee) [reply tests/keras/applications/applications_test.py]: I agree. How about removing `set_image_data_format` @myutwo150? Other application tests except `test_inceptionresnetv2_notop` do not include `set_image_data_format`.

**Comment by myutwo150** [comment]
There's still a risk that `image_data_format` is not set back to its original value in `test_inceptionresnetv2_notop`. Maybe we can just remove the test about `'channels_first'` from this function (like the other application `test_*_no_top` tests)?

Other than that, LGTM. What do you think? @taehoonlee @fchollet

File: tests/keras/applications/applications_test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (myutwo150) [reply tests/keras/applications/applications_test.py]: Sure. Sounds good to me.

**Comment by myutwo150** [comment]
LGTM.

**Comment by taehoonlee** [comment]
@fchollet, We think this PR is ready to merge.

**Pull request merged**